plugins/experimental/server_push_preload/server_push_preload.cc

```diff
@@ -0,0 +1,109 @@
+#include <iostream>
+#include <regex>
+#include <sstream>
+#include <vector>
+#include <ts/ts.h>
+#include <ts/experimental.h>
+#include <atscppapi/GlobalPlugin.h>
+#include <atscppapi/utils.h>
+
+#define PLUGIN_NAME "server_push_preload"
+
+using namespace std;
+using namespace atscppapi;
+
+static regex linkRegexp("<([^>]+)>;.*rel=preload");
+
+namespace
+{
+GlobalPlugin *plugin;
+}
+
+class LinkServerPushPlugin : public GlobalPlugin
+{
+public:
+  LinkServerPushPlugin()
+  {
+    TSDebug(PLUGIN_NAME, "registering transaction hooks");
+    LinkServerPushPlugin::registerHook(HOOK_SEND_RESPONSE_HEADERS);
+  }
+
+  virtual void
+  handleSendResponseHeaders(Transaction &transaction)
+  {
+    serverPush(transaction);
+    transaction.resume();
+  }
+
+  void
+  serverPush(Transaction &transaction)
+  {
+    ClientRequest &request = transaction.getClientRequest();
+    if (request.getMethod() != HTTP_METHOD_GET) {
```

Comment:
I have been using this one - https://docs.trafficserver.apache.org/en/latest/developer-guide/api/functions/TSClientProtocolStack.en.html

That should work. right?